DogStatsd C#: Add section Client-side aggregation (#12428)
DataDog/documentation: content/en/developers/dogstatsd/high_throughput.md

> Client libraries can aggregate metrics on the client side, reducing number of messages that have to be submitted to the DataDog Agent, improving IO performance and throughput.

Reviewer: ```suggestion
Client libraries can aggregate metrics on the client side, reducing number of messages that have to be submitted to the Datadog Agent, improving IO performance and throughput.
```

> {{< programming-lang-wrapper langs="go,java,csharp" >}}

Reviewer: ```suggestion
{{< programming-lang-wrapper langs="go,java,.NET" >}}
```

[1]: https://github.com/DataDog/java-dogstatsd-client#aggregation
{{< /programming-lang >}}

{{< programming-lang lang="csharp" >}}

> Client-side aggregation is available in dogstatsd-csharp-client version 7.0.0 and later, and is enabled by default.

Reviewer: ```suggestion
Client-side aggregation is available in dogstatsd-csharp-client v7.0.0+ and is enabled by default.
```

```csharp
var dogstatsdConfig = new StatsdConfig
{
    StatsdServerName = "127.0.0.1",
    StatsdPort = 8125,
    ClientSideAggregation = new ClientSideAggregationConfig()
};
```

> Client-side aggregation is available for gauges, counters and sets.

Reviewer: ```suggestion
Client-side aggregation is available for gauges, counters, and sets.
```

> See [Client-side aggregation][https://github.com/DataDog/dogstatsd-csharp-client#client-side-aggregation] for more information.

Reviewer: ```suggestion
For more information, see the [DogStatsD for C# repository][1].
```

> {{< /programming-lang >}}

Reviewer: ```suggestion
[1]: https://github.com/DataDog/dogstatsd-csharp-client#client-side-aggregation
{{< /programming-lang >}}
```

[1]: https://github.com/DataDog/java-dogstatsd-client#aggregation
{{< /programming-lang >}}

> {{< programming-lang lang="csharp" >}}

Reviewer: ```suggestion
{{< programming-lang lang=".NET" >}}
```

> Client-side aggregation is available in dogstatsd-csharp-client v7.0.0+ and is enabled by default.

Reviewer: ```suggestion
Client-side aggregation is available in DogStatsD C# client v7.0.0+ and is enabled by default.
```

```csharp
var dogstatsdConfig = new StatsdConfig
{
    StatsdServerName = "127.0.0.1",
    StatsdPort = 8125,
    ClientSideAggregation = new ClientSideAggregationConfig()
};
```

> Client-side aggregation is available for gauges, counters, and sets.

Reviewer: ```suggestion
```
